service/service.go

```diff
@@ -138,3 +150,37 @@ func (s *Service) Boot() {
 		go s.KVMClusterConfigFramework.Boot()
 	})
 }
+
+func newBaseClusterConfig(f *flag.Flag, v *viper.Viper) (*cluster.Config, error) {
+	guestClusterCIDR, _, err := net.ParseCIDR(v.GetString(f.Guest.Cluster.Kubernetes.API.ClusterIPRange))
+	if err != nil {
+		return nil, microerror.Maskf(invalidConfigError, "invalid Kubernetes ClusterIPRange")
+	}
+
+	guestClusterAPIServerIP, err := newAPIServerIP(guestClusterCIDR)
+
+	clusterConfig := &cluster.Config{
+		CertTTL: v.GetString(f.Guest.Cluster.Vault.Certificate.TTL),
+		IP: cluster.IP{
+			API:   guestClusterAPIServerIP,
+			Range: guestClusterCIDR,
+		},
+	}
+
+	return clusterConfig, nil
+}
+
+func newAPIServerIP(ip net.IP) (net.IP, error) {
+	ip = ip.To4()
+	if ip != nil {
+		return nil, microerror.Maskf(invalidConfigError, "Kubernetes ClusterIPRange CIDR must be an IPv4 range")
+	}
+
+	// IP must be a network address
+	if ip[3] != 0 {
+		return nil, microerror.Maskf(invalidConfigError, "Kubernetes ClusterIPRange CIDR must be a network address")
+	}
```

Comment:
This assumes the cidr mask is `<= 24`, doesn't it? Shouldn't it be checked first?

Reply:
Yes this does, but is there any real world scenario that mask would be > 24? Kubernetes design (at least originally) assigns /24 for each node. That idea would break if mask > 24. WDYT?

Reply:
I have no fucking clue about this here. :D

Reply:
I suggest checking if `mask <= 24` and return invalidConfigError if not. Does that make sense?

Reply:
Refactored parsing to be more robust + added tests.